File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/cadastro/usuario/UsuarioService.java
package br.com.grupopipa.gestaointegrada.cadastro.usuario;

import br.com.grupopipa.gestaointegrada.core.service.CrudService;

public interface UsuarioService extends CrudService<UsuarioDTO, UsuarioGridDTO> {

    UsuarioDTO findUsuarioDTOByLogin(String login);
    
}

File: src/frontend/src/app/services/usuario.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from './base-service';
import { HttpClient } from '@angular/common/http';
import { UsuarioDTO } from '../model/usuario-dto';
import { UsuarioGridDTO } from '../model/usuario-grid-dto';
import { UsuarioBackendMessages } from '../components/cadastro/usuario/usuario-backend-message.service';
import { MessageService } from '../components/base/messages/messages.service';

@Injectable()
export class UsuarioService extends BaseService<UsuarioGridDTO, UsuarioDTO> {
  private static readonly USUARIO = 'usuario';

  constructor(httpClient: HttpClient, messageService: MessageService, backendMessages: UsuarioBackendMessages) {
    super(httpClient, UsuarioService.USUARIO, messageService, backendMessages);
  }
}
